### PR TITLE
Added tooltip when copy invite URL to clipboard

### LINKED ISSUE #1455: Copy to clipboard button needs tooltip when clicked
Screenshot 2020-12-02 at 12 01 51

Also, "pressed" state isn't needed here. Right now it turns gray when pressed

### PR DESCRIPTION
**Pull request by staked-smart-ace**
Closes #1455

### CODE REVIEW

File: ui/app/AppLayouts/Chat/components/InviteFriendsPopup.qml [hunk 1]
```diff
@@ -21,7 +21,8 @@ ModalPopup {
         anchors.left: linkText.right
         anchors.leftMargin: Style.current.smallPadding
         anchors.verticalCenter: linkText.verticalCenter
-        textToCopy: popup.getStatusText
+        textToCopy: popup.getStatusText.substr(popup.getStatusText.indexOf("https"))
+        textTooltip: "Copied \"" + popup.getStatusText.substr(popup.getStatusText.indexOf("https")) + "\" to clipboard"
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/InviteFriendsPopup.qml:25]: To enable i18n, we need to use `qsTr`. It's very simple, but I don't blame you for not knowing about it.
Here is the documentation if you want to learn about it: https://doc.qt.io/qt-5/qtquick-internationalization.html
```suggestion
        textTooltip: qsTr('Copied "%1" to clipboard').arg(popup.getStatusText.substr(popup.getStatusText.indexOf("https")))
```

File: ui/shared/CopyToClipBoardButton.qml [hunk 2]
```diff
@@ -44,6 +47,31 @@ Rectangle {
...
+        text: parent.textTooltip
+        parent: copyToClipboardButton
+    }
+
+    Timer {
+        id:showTimer
+        interval: 500
+        running: parent.showTooltip && !toolTip.visible
+        onTriggered: {
+            toolTip.visible = true;
```
Comment (jrainville) [inline ui/shared/CopyToClipBoardButton.qml:62]: I think you can set the tooltip to `visible` straight away and not need the 500 timer. Unless you tell me it was too ugly?

File: ui/shared/CopyToClipBoardButton.qml [hunk 3]
```diff
@@ -44,6 +47,31 @@ Rectangle {
...
+        }
+    }
+
+    Timer {
+        id:hideTimer
+        interval: 1500
```
Comment (jrainville) [inline ui/shared/CopyToClipBoardButton.qml:68]: I feel like `1500` might be too quick to read?

**Review by jrainville — COMMENTED** [review]
Nice PR. I added a suggestion to add i18n and also asked a question about the timers

File: ui/shared/CopyToClipBoardButton.qml (hunk 3 above)
Comment (staked-smart-ace) [reply ui/shared/CopyToClipBoardButton.qml:68]: what about 2 or 2.5 sec?

File: ui/shared/CopyToClipBoardButton.qml (hunk 2 above)
Comment (staked-smart-ace) [reply ui/shared/CopyToClipBoardButton.qml:62]: Generally, tooltip has a little delay before it is shown.
If you think 0.5 sec is too long, let me update more short delay like 0.2 or 0.3

File: ui/shared/CopyToClipBoardButton.qml (hunk 3 above)
Comment (jrainville) [reply ui/shared/CopyToClipBoardButton.qml:68]: I'd need to test for myself, but it's surely better. Try to find an amount that leaves enough time to read the text

File: ui/app/AppLayouts/Chat/components/InviteFriendsPopup.qml (hunk 1 above)
Comment (staked-smart-ace) [reply ui/app/AppLayouts/Chat/components/InviteFriendsPopup.qml]: I just updated as you mentioned.

**Review by jrainville — APPROVED** [review]
Well played

**Comment by PascalPrecht** [comment]
Hey @staked-smart-ace !
Thanks for your contribution. Unfortunately, I didn't get a chance to review this before it got merged.
There's a few things that need to be addressed:

1. We already have a `StatusToolTip` component that should be used instead of the vanilla `ToolTip` component. You can find it in `ui/shared/status`
2. There are other places in the application where `CopyToClipBoardButton` is used but don't have a tooltip text. This now generates a broken UI as shown in the screenshot below:

Screenshot 2021-01-13 at 10 16 39


Would you mind sending a PR that addresses these points?

**Comment by PascalPrecht** [comment]
Ha, funny. The screenshot doesn't show the tooltip... well, try it yourself, it creates an empty tooltip

**Comment by staked-smart-ace** [comment]
> Ha, funny. The screenshot doesn't show the tooltip... well, try it yourself, it creates an empty tooltip

So, what is your wanted solution for this issue?
Do you want to show default copied text or don't show tooltip if not inputed tooltip text param?

**Comment by PascalPrecht** [comment]
Ah sorry for not being clear here. As discussed on Status, it should actually always say "Copied!" 
No need to pass down text from the outside.

**Comment by staked-smart-ace** [comment]
I updated Tooltip with StatusTooltip and made new PR.
Please review this.
https://github.com/status-im/nim-status-client/pull/1652

**Pull request merged**